From 4a8ccaf76dbd978be646e8b6818fb4cfdf7fbacd Mon Sep 17 00:00:00 2001
From: Liu Ying <b17645@freescale.com>
Date: Thu, 28 Jan 2010 10:19:23 -0500
Subject: [PATCH] ENGR00120500 V4L2 overlay:Unblank overlay fb before enable PRP_VF channel

This change makes us get rid of DIx_SYNC_DISP_ERR. The error sometimes
causes the backgound framebuffer reinitialized automatically, even if
we preview captured video on overlay framebuffer.

Signed-off-by: Liu Ying <b17645@freescale.com>
---
 drivers/media/video/mxc/capture/ipu_prp_vf_sdc.c |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/media/video/mxc/capture/ipu_prp_vf_sdc.c b/drivers/media/video/mxc/capture/ipu_prp_vf_sdc.c
index 75574e9..9994012 100644
--- a/drivers/media/video/mxc/capture/ipu_prp_vf_sdc.c
+++ b/drivers/media/video/mxc/capture/ipu_prp_vf_sdc.c
@@ -96,6 +96,10 @@ static int prpvf_start(void *private)
 	ipu_disp_set_window_pos(MEM_FG_SYNC, cam->win.w.left,
 			cam->win.w.top);
 
+	acquire_console_sem();
+	fb_blank(fbi, FB_BLANK_UNBLANK);
+	release_console_sem();
+
 	memset(&vf, 0, sizeof(ipu_channel_params_t));
 	ipu_csi_get_window_size(&vf.csi_prp_vf_mem.in_width,
 				&vf.csi_prp_vf_mem.in_height, cam->csi);
@@ -253,10 +257,6 @@ static int prpvf_start(void *private)
 		ipu_select_buffer(CSI_PRP_VF_MEM, IPU_OUTPUT_BUFFER, 1);
 	}
 
-	acquire_console_sem();
-	fb_blank(fbi, FB_BLANK_UNBLANK);
-	release_console_sem();
-
 	cam->overlay_active = true;
 	return err;
 
-- 
1.5.4.4

